### ISSUE TITLE
Incorrect loop when packing ROM class on JITServer client

### ISSUE THREAD

**Issue by dmitry-ten** [eclipse-openj9/openj9#13028]
A bunch of tests in nightly runs fail due to hangs.
I was able to reproduce the hang on ppc64le, with `DaaLoadTest_daa3_special_5m_14` test.

Attaching gdb to the client's process revealed a compilation that hanged while handling `ResolvedMethod_getRemoteROMClassAndMethods` message.
The stack trace showed the following:
```
(gdb) bt
#0  allSlotsInBytecodesDo (userData=0x71fb10f96f80, callbacks=0x71fb10f96e78, method=0x71fa94011974, romClass=0x71fa940074f0)
    at /root/src-11/openj9-openjdk-jdk11/openj9/runtime/util/romclasswalk.c:583
#1  allSlotsInROMMethodDo (userData=0x71fb10f96f80, callbacks=0x71fb10f96e78, method=0x71fa94011974, romClass=0x71fa940074f0)
    at /root/src-11/openj9-openjdk-jdk11/openj9/runtime/util/romclasswalk.c:278
#2  allSlotsInROMMethodsSectionDo (romClass=0x71fa940074f0, callbacks=0x71fb10f96e78, userData=0x71fb10f96f80)
    at /root/src-11/openj9-openjdk-jdk11/openj9/runtime/util/romclasswalk.c:226
#3  0x000071fb128351c4 in allSlotsInROMClassDo (romClass=0x71fa940074f0, callback=<optimized out>, sectionCallback=<optimized out>,
    validateRangeCallback=<optimized out>, userData=0x71fb10f96f80) at /root/src-11/openj9-openjdk-jdk11/openj9/runtime/util/romclasswalk.c:207
#4  0x000071fb11e5ee48 in JITServerHelpers::packROMClass (romClass=0x71fa940074f0, trMemory=0x71fb10f9a9c8, packedSize=@0x71fb10f970c8: 130456,
    expectedSize=0) at /root/src-11/openj9-openjdk-jdk11/openj9/runtime/compiler/control/JITServerHelpers.cpp:314
#5  0x000071fb11e67668 in JITServerHelpers::packRemoteROMClassInfo[abi:cxx11](J9Class*, J9VMThread*, TR_Memory*, bool) (clazz=0x4e3800,
    vmThread=<optimized out>, trMemory=0x71fb10f9a9c8, serializeClass=<optimized out>)
    at /root/src-11/openj9-openjdk-jdk11/openj9/runtime/compiler/control/JITServerHelpers.cpp:640
#6  0x000071fb11e1cbe4 in handleServerMessage (client=0x71faf0ea3b20, fe=0x71faec007c40,
    response=@0x71fb10f99b86: JITServer::ResolvedMethod_getRemoteROMClassAndMethods)
    at /root/src-11/openj9-openjdk-jdk11/openj9/runtime/compiler/control/JITClientCompilationThread.cpp:1151
```
This is the code around `romclasswalk.c:583`.
```
 582                                 for (i = 0; i <= (U_32)(high - low); ++i) {
 583                                         callbacks->slotCallback(romClass, J9ROM_U32, pc, "bcArg32", userData);
 584                                         pc += 4;
 585                                 }
```
Checked the value of `i`:
```
(gdb) p i
$1 = 1319276594
```
Such a large value indicates that `(high-low)` must be a negative value that was casted to an unsigned integer resulting in an extremely large number, producing a near-infinite loop.

**Comment by dmitry-ten** [comment]
@mpirvu fyi.
@AlexeyKhrabrov since you worked on the new code for ROM class packing, do you have any advice on what this code does and what could be causing `low` and `high` to be incorrect?

**Comment by dmitry-ten** [comment]
The name of the class being packed: `net/openj9/test/arithmetics/TestArithmeticOperations&`

**Comment by dmitry-ten** [comment]
So for a `JBtableswitch`, `low` and `high` are supposed to be arguments representing the lowest and highest case numbers in a switch statement, that has integers as cases. Still don't know why these values would be invalid though when we pack a ROM class on the client.

**Comment by AlexeyKhrabrov** [comment]
Either the code in `romclasswalk.c` is broken, or the ROMClass in question is malformed for some reason. I don't see any recent changes to the ROMClass format other than some value types stuff (which is probably unrelated to this), so I don't know what could have broken it. Did this test pass before? Maybe this specific class wasn't touched by the JIT before.

This is a problem regardless of JITServer. The same code is used by cfdumper, and there is another equivalent version of the ROMClass walk in DDR. The loop in question looks the same there:
https://github.com/eclipse-openj9/openj9/blob/629fc4f8eb18604d91335f8fb73e323d953f8eee/debugtools/DDR_VM/src/com/ibm/j9ddr/vm29/tools/ddrinteractive/RomClassWalker.java#L534-L539

We can work around this for JITServer. We don't actually need to visit the vast majority of ROMClass slots since we only care about UTF8 and NAS slots. We can add a parameter to `allSlotsInROMClassDo()` to skip everything except UTF8 and NAS, or at least to skip visiting bytecodes. This would also reduce the overhead of packing. Another option is to implement a separate walk that only visits UTF8 and NAS slots. The only problem with these approaches is increasing complexity and maintenance burden.

**Comment by dmitry-ten** [comment]
I was also able to reproduce the same issue when packing ROM class for a compilation request, so that makes ROM class being malformed unlikely, since it's not even requested by the server at this point. 
I expect that the issue is with the bytecode walk somehow. Maybe the bytecode walk doesn't increment pc by the correct amount for some previous bytecode, resulting in an incorrect bytecode index being processed.

**Comment by dmitry-ten** [comment]
Seems to be the case, comparing bytecode iterator in `RomClassWalker.java` to the one in `romclasswalk.c`, the latter is missing handlers for 2 bytecodes: `JBwithfield` and `JBdefaultvalue`.

**Comment by dmitry-ten** [comment]
Fixing this doesn't fix the bug however. Makes sense, since the missing bytecodes are value type bytecodes, which shouldn't be enabled by default for JDK11. 
Maybe some bytecodes are missing in both iterators, will check that.

**Comment by dmitry-ten** [comment]
I reasoned that if the bytecode iterator is missing a handler for some of the bytecodes, then the bytecode before `JBtableswitch` is likely the one not handled. Taking `(pc-1)` at the beginning of `JBtableswitch` handler will produce the bytecode index of a missing handler.
Indeed, the previous bytecode index was equal to `JBinvokedynamic`, which is not handled by the function. Adding a case for it fixed the bug, making the test pass.
I also found out that a handler for `JBinvokeinterface` was missing as well. 

Made a PR to fix the issue: #13035
